graphics/fonts/bdf.cpp

```diff
@@ -787,59 +797,76 @@ BdfFont *BdfFont::scaleFont(const BdfFont *src, int newSize) {
 		const BdfBoundingBox &box = data.boxes ? data.boxes[i] : data.defaultBox;
 		const BdfBoundingBox &srcBox = data.boxes ? src->_data.boxes[i] : src->_data.defaultBox;
 
+		int grayLevel = box.height * box.width / 3;
+		int dstPitch = (box.width + 7) / 8 ;
+
+#if DRAWDEBUG
+		int ccc = 'a';
+#endif
 		if (src->_data.bitmaps[i]) {
-			const int bytes = ((box.width + 7) / 8) * box.height; // Dimensions have been already corrected
+			int *dstGrayPointer = 0;
+			Graphics::Surface *srcSurf = scaledFont->scaleSingleGlyph(box.width, box.height, grayLevel, i + src->_data.firstCharacter, 
+																  	  box.xOffset, src->_data.height, srcBox.width, scale, &dstGrayPointer);
```

Comment:
Okay, now you allocate dstGrayPointer in `scaleSingleGlyph()`. Where do you free this memory?

Reply:
it is happening in the deconstructor

Reply:
sorry I missunderstood, I have replace it to dstGra, which is required for debuging